plugins/slo/pkg/slo/interfaces.go

```diff
@@ -26,25 +28,50 @@ type SLOStore interface {
 	Status(existing *sloapi.SLOData) (*sloapi.SLOStatus, error)
 	WithCurrentRequest(req proto.Message, ctx context.Context) SLOStore
 }
+type ServiceBackend interface {
+	List(clusters *corev1.ClusterList) (*sloapi.ServiceList, error)
+	GetMetricId() (*MetricIds, error)
+	WithCurrentRequest(req proto.Message, ctx context.Context) ServiceBackend
+}
+
+type MetricIds struct {
+	Good  string
+	Total string
+}
 
-type SLORequestBase struct {
+type RequestBase struct {
 	req proto.Message
 	p   *Plugin
 	ctx context.Context
 	lg  hclog.Logger
 }
 
 type SLOMonitoring struct {
-	SLORequestBase
+	RequestBase
 }
 
 type SLOLogging struct {
-	SLORequestBase
+	RequestBase
+}
+
+type MonitoringServiceBackend struct {
+	RequestBase
 }
 
 func NewSLOMonitoringStore(p *Plugin, lg hclog.Logger) SLOStore {
 	return SLOMonitoring{
-		SLORequestBase{
+		RequestBase{
+			req: nil,
+			p:   p,
+			ctx: context.TODO(),
```

Comment:
In general, context.Background() is better unless you actually need to come back to this (if that is the case, ignore this comment)

Reply:
Changed to context.Background()